[Documentation]

### Wide character type PyTorch Inference crash

I am using ONNX Runtime C++ pre-built version 1.14.0 and following along with tutorial on PyTorch Inference. While I'm trying to build inference example I'm getting error using wide character type, line:

`Ort::Session session{env, L"model.onnx", Ort::SessionOptions{nullptr}};`

Crashing with error: 
```
error: no matching function for call to ‘Ort::Session::Session(<brace-enclosed initializer list>)’
   14 | v, L"/home/novarc/onnxruntime-example/resnest_50_new_aug_ur_1.onnx", Ort::SessionOptions{nullptr}};
      |                                                                                                  ^

In file included from /opt/onnxruntime/include/onnxruntime_cxx_api.h:1876,
                 from /home/novarc/onnxruntime-example/main.cpp:5:
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:950:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const void*, size_t, const Ort::SessionOptions&, OrtPrepackedWeightsContainer*)’
  950 | inline Session::Session(const Env& env, const void* model_data, size_t model_data_length,
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:950:8: note:   candidate expects 5 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:946:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const void*, size_t, const Ort::SessionOptions&)’
  946 | inline Session::Session(const Env& env, const void* model_data, size_t model_data_length, const SessionOptions& options) {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:946:8: note:   candidate expects 4 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:941:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const char*, const Ort::SessionOptions&, OrtPrepackedWeightsContainer*)’
  941 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options,
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:941:8: note:   candidate expects 4 arguments, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:937:8: note: candidate: ‘Ort::Session::Session(const Ort::Env&, const char*, const Ort::SessionOptions&)’
  937 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options) {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_inline.h:937:58: note:   no known conversion for argument 2 from ‘const wchar_t [62]’ to ‘const char*’
  937 | inline Session::Session(const Env& env, const ORTCHAR_T* model_path, const SessionOptions& options) {
      |                                         ~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /home/novarc/onnxruntime-example/main.cpp:5:
/opt/onnxruntime/include/onnxruntime_cxx_api.h:783:12: note: candidate: ‘Ort::Session::Session(std::nullptr_t)’
  783 |   explicit Session(std::nullptr_t) {}                                                   ///< Create an empty Session object, must be assigned a valid one to be used
      |            ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_api.h:783:12: note:   candidate expects 1 argument, 3 provided
/opt/onnxruntime/include/onnxruntime_cxx_api.h:782:8: note: candidate: ‘Ort::Session::Session(Ort::Session&&)’
  782 | struct Session : detail::SessionImpl<OrtSession> {
      |        ^~~~~~~
/opt/onnxruntime/include/onnxruntime_cxx_api.h:782:8: note:   candidate expects 1 argument, 3 provided
make[2]: *** [CMakeFiles/onnxruntime-example.dir/build.make:76: CMakeFiles/onnxruntime-example.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/onnxruntime-example.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

If I'm changing this line to: `Ort::Session session{env, "model.onnx", Ort::SessionOptions{nullptr}};` everything works fine.

### Page / URL

https://onnxruntime.ai/docs/tutorials/accelerate-pytorch/pytorch.html

--- a/docs/execution-providers/OpenVINO-ExecutionProvider.md
+++ b/docs/execution-providers/OpenVINO-ExecutionProvider.md
@@ -170,7 +170,7 @@ Use the ONNX API's.[documentation](https://github.com/onnx/onnx/blob/master/docs
 
 Example:
 
-```bash
+```python
 import onnx
 onnx_model = onnx.load("model.onnx") # Your model in memory as ModelProto
 onnx.save_model(onnx_model, 'saved_model.onnx', save_as_external_data=True, all_tensors_to_one_file=True, location='data/weights_data', size_threshold=1024, convert_attribute=False)

--- a/docs/performance/graph-optimizations.md
+++ b/docs/performance/graph-optimizations.md
@@ -135,11 +135,11 @@ session = rt.InferenceSession("<model_path>", sess_options)
   g_ort->SetSessionGraphOptimizationLevel(session_options, ORT_ENABLE_EXTENDED);
 
   // To enable model serialization after graph optimization set this
-  const wchar_t* optimized_model_path = L"optimized_model_path";
+  const ORTCHAR_T* optimized_model_path = ORT_TSTR("optimized_model_path");
   g_ort->SetOptimizedModelFilePath(session_options, optimized_model_path);
 
   OrtSession* session;
-  const wchar_t* model_path = L"model_path";
+  const ORTCHAR_T* model_path = ORT_TSTR("model_path");
   g_ort->CreateSession(env, model_path, session_option, &session);
 ```
 

--- a/docs/performance/tune-performance.md
+++ b/docs/performance/tune-performance.md
@@ -531,7 +531,7 @@ api.SessionOptionsAppendExecutionProvider_CUDA_V2(static_cast<OrtSessionOptions*
 
 
 // Create IO bound inputs and outputs.
-Ort::Session session(*ort_env, L"matmul_2.onnx", session_options);
+Ort::Session session(*ort_env, ORT_TSTR("matmul_2.onnx"), session_options);
 Ort::MemoryInfo info_cuda("Cuda", OrtAllocatorType::OrtArenaAllocator, 0, OrtMemTypeDefault);
 Ort::Allocator cuda_allocator(session, info_cuda);
 

--- a/docs/reference/operators/add-custom-op.md
+++ b/docs/reference/operators/add-custom-op.md
@@ -94,7 +94,7 @@ Ort::SessionOptions session_options;
 session_options.Add(domain);  // Add the domain to the session options.
 
 // Create a session.
-Ort::Session session(env, "my_model_with_custom_ops.onnx", session_options);
+Ort::Session session(env, ORT_TSTR("my_model_with_custom_ops.onnx"), session_options);
 ```
 ## Create a library of custom operators
 Custom operators can be defined in a separate shared library (e.g., a .dll on Windows or a .so on Linux). A custom operator library must export and implement a `RegisterCustomOps` function. The `RegisterCustomOps` function adds a `Ort::CustomOpDomain` containing the library's custom operators to the provided session options.
@@ -183,9 +183,9 @@ Once compiled, the custom operator shared library can then be registered with an
 Ort::Env env;
 Ort::SessionOptions session_options;
 
-session_options.RegisterCustomOpsLibrary(L"my_custom_op.dll");
+session_options.RegisterCustomOpsLibrary_V2(ORT_TSTR("my_custom_op.dll"));
 
-Ort::Session session(env, "my_model.onnx", session_options);
+Ort::Session session(env, ORT_TSTR("my_model.onnx"), session_options);
 ```
 
 ### Examples
@@ -300,7 +300,7 @@ custom_op_configs.AddConfig("OpenVINO_Wrapper", "device_type", "CPU");
 // Register custom op library and pass in the custom op configs (optional).
 session_options.RegisterCustomOpsLibrary("MyOpenVINOWrapper_Lib.so", custom_op_configs);
 
-Ort::Session session(env, "custom_op_mnist_ov_wrapper.onnx", session_options);
+Ort::Session session(env, ORT_TSTR("custom_op_mnist_ov_wrapper.onnx"), session_options);
 ```
 
 Refer to the [complete OpenVINO custom operator wrapper example](https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/test/testdata/custom_op_openvino_wrapper_library) for more details. To create an ONNX model that wraps an external model or weights, refer to the [create_custom_op_wrapper.py tool](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/custom_op_wrapper/create_custom_op_wrapper.py).

--- a/docs/tutorials/accelerate-pytorch/pytorch.md
+++ b/docs/tutorials/accelerate-pytorch/pytorch.md
@@ -157,7 +157,7 @@ Then you can run the ONNX model in the environment of your choice. The ONNXRunti
   // Allocate ONNXRuntime session
   auto memory_info = Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
   Ort::Env env;
-  Ort::Session session{env, L"model.onnx", Ort::SessionOptions{nullptr}};
+  Ort::Session session{env, ORT_TSTR("model.onnx"), Ort::SessionOptions{nullptr}};
 
   // Allocate model inputs: fill in shape and size
   std::array<float, ...> input{};

--- a/docs/tutorials/mnist_cpp.md
+++ b/docs/tutorials/mnist_cpp.md
@@ -128,13 +128,13 @@ To make things more interesting, the window painting handler graphs the probabil
 ### The Ort::Session
 
 1. Creation: The Ort::Session is created inside the MNIST structure here:
-     ```
-    Ort::Session session_{env, L"model.onnx", Ort::SessionOptions{nullptr}};
+     ```c++
+    Ort::Session session_{env, ORT_TSTR("model.onnx"), Ort::SessionOptions{nullptr}};
     ```
     [[Source]](https://github.com/microsoft/onnxruntime/blob/521dc757984fbf9770d0051997178fbb9565cd52/samples/c_cxx/MNIST/MNIST.cpp#L43)
 
 2. Setup inputs & outputs: The input & output tensors are created here:
-    ```
+    ```c++
     MNIST() {
     auto allocator_info = Ort::AllocatorInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
     input_tensor_ = Ort::Value::CreateTensor<float>(allocator_info, input_image_.data(), input_image_.size(), input_shape_.data(), input_shape_.size());
@@ -146,7 +146,7 @@ To make things more interesting, the window painting handler graphs the probabil
     In this usage, we're providing the memory location for the data instead of having Ort allocate the buffers. This is simpler in this case since the buffers are small and can just be fixed members of the MNIST struct.
 
 3. Run: Running the session is done in the Run() method:
-    ```
+    ```c++
      int Run() {
     const char* input_names[] = {"Input3"};
     const char* output_names[] = {"Plus214_Output_0"};

